python/lsst/metaserv/schemaToMeta.py

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2015 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+
+import os
+import pprint
+import re
+import sys
+
+class SchemaToMeta(object):
+    """
+    SchemaToMeta class parses mysql schema file that can optionally contain
+    extra tokens in comments. Extracts information for each table:
+    * name
+    * engine
+    * per column:
+      - type
+      - notnull
+      - defaultValue
+      - <descr>...</descr>
+      - <unit>...</unit>
+      - <ucd>...</ucd>
+    and saves it in an array.
+
+    This code was originally written for schema browser
+    (in cat/bin/schema_to_metadata.py).
+    """
+
+    def __init__(self, inputFileName):
+        """
+        Constructor, prepares for execution.
+
+        @param inputFileName:   ASCII file containing the schem to be parsed
+        """
+        if not os.path.isfile(inputFileName):
+            sys.stderr.write("File '%s' does not exist\n" % inputFileName)
+            sys.exit(1)
+        self._inFName = inputFileName
+
+        self._tableStart = re.compile(r'CREATE TABLE (\w+)*')
+        self._tableEnd = re.compile(r"\)")
+        self._engineLine = re.compile(r'\) (ENGINE|TYPE)=(\w+)*;')
+        self._columnLine = re.compile(r'[\s]+(\w+) ([\w\(\)]+)')
+        self._unitLine = re.compile(r'<unit>(.+)</unit>')
+        self._ucdLine = re.compile(r'<ucd>(.+)</ucd>')
+        self._descrLine = re.compile(r'<descr>(.+)</descr>')
+        self._descrStart = re.compile(r'<descr>(.+)')
+        self._descrMiddle = re.compile(r'\s*--(.+)')
+        self._descrEnd = re.compile(r'\s*--(.+)</descr>')
```

Comment:
I'd go one step further and made them class variables, there is no reason they have to be per-instance variables.